### ISSUE TITLE
[BUG] (Including PreQuantile) TabularEnsemble in timeseries prediction cannot handle known covariates

### ISSUE THREAD

**Issue by RDR2Blackwater** [autogluon/autogluon#5523]
**Bug Report Checklist**

- [√] I provided code that demonstrates a minimal reproducible example.  
- [ ] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [√] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**
As description, TabularEnsemble and PreQuantileTabularEnsemble cannot handle known covariates correctly, since the `fit` method directly reshape the whole label, including known covariates.

**To Reproduce**
```python
import os
import os.path
# Set envs before import autogluon
os.environ["HF_HOME"] = os.path.abspath("./huggingface")  # Use model weight and configs inside this project
os.environ['HF_HUB_OFFLINE'] = "1"  # Set to "1" to disable huggingface warning, if you already buffered the model weight

import copy
import numpy as np
import scipy.stats as stats
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib.patches as mpatches

from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
from typing import Any, Optional, Union

from autogluon.timeseries import TimeSeriesPredictor, TimeSeriesDataFrame
from autogluon.common.utils.utils import setup_outputdir
from autogluon.core.utils.loaders import load_pkl
from autogluon.core.utils.savers import save_pkl

# Basic hyperparameters during model initialization
version_name = "TabularEnsembleTest"
eval_metric = "RMSE"

# Basic hyperparameters during training
backtesting_times_in_val = 5
keep_logs = True
enable_ensemble = True

# Basic hyperparameters during testing/inferencing
quantile_levels = [0.1, 0.9]
test_history_length = 18

hyperparameters = {
    # ---------------------- Seasonality-specified models ----------------------
    "NPTS":
        {
            "use_seasonal_model": False
        },
    "AutoARIMA":
        {
            "seasonal_period": 1,
            "d": 1,
            "start_p": 1,
            "start_q": 1,
        },
}

ensemble_hyperparameters = {"TabularEnsemble": {}}  # Specified tabular ensemble

# Custom multivariate prediction class
class MultiTargetTimeSeriesPredictor:
    """
    A predictor for multi-target time series forecasting.

    This wraps multiple TimeSeriesPredictor objects, one for each target.
    It mimics the API of the Tabular MultiTargetPredictor but is adapted for time series.
    The underlying predictors are held in memory during an interactive session but are
    persisted via their paths to manage resources, following AutoGluon's best practices.

    Parameters
    ----------
    target_columns : List[str]
        A list of column names to be predicted. Renamed from 'labels' for clarity.
    path : str, default = None
        Path to the directory where models and intermediate outputs will be saved.
    eval_metric : str or Dict[str, str], optional
        Evaluation metric for each target. If a string, used for all targets.
        If a dict, specifies metric per target. Defaults to AutoGluon's time series default ('MASE').
    predictor_init_kwargs : Dict[str, Any], optional
        Additional keyword arguments passed to the constructor of each individual TimeSeriesPredictor.
        For example: `dict(prediction_length=24)`.
    """
    _PREDICTOR_FILENAME = "MTS_predictor.pkl"

    def __init__(
        self,
        target_columns: list[str],
        path: str,
        eval_metric: Optional[Union[str, dict[str, str]]] = None,
        predictor_init_kwargs: Optional[dict[str, Any]] = None,
    ):
        if not isinstance(target_columns, list) or len(target_columns) < 2:
            raise ValueError("MultiTargetTimeSeriesPredictor requires a list of at least two target columns.")

        self.path = setup_outputdir(path, warn_if_exist=False)
        self.target_columns = target_columns
        self.predictor_init_kwargs = predictor_init_kwargs if predictor_init_kwargs else {}
        self.eval_metric = eval_metric
        self.predictors = {}

        for target in self.target_columns:
            predictor_path = os.path.join(self.path, f"Predictor_{target}")

            current_eval_metric = self.eval_metric
            if isinstance(self.eval_metric, dict):
                current_eval_metric = self.eval_metric.get(target, "RMSE")  # Use RMSE by default.

            self.predictors[target] = TimeSeriesPredictor(
                target=target,
                path=predictor_path,
                eval_metric=current_eval_metric,
                **self.predictor_init_kwargs,
            )


    def fit(self, train_data: TimeSeriesDataFrame, **fit_kwargs):
        """
        Fits a separate TimeSeriesPredictor for each target.
        """
        for i, target in enumerate(self.target_columns):
            print(f"[{i+1}/{len(self.target_columns)}] Fitting predictor for target: {target} ...")
            predictor = self.get_predictor(target)

            # Prevent data leakage by removing other target columns from the training data
            other_targets = [t for t in self.target_columns if t != target]
            current_train_data = train_data.drop(columns=other_targets)

            predictor.fit(train_data=current_train_data, **fit_kwargs)
        # Save fitted models
        self.save()


    def predict(
        self,
        data: TimeSeriesDataFrame,
        known_covariates: Optional[TimeSeriesDataFrame] = None,
        **predict_kwargs
    ) -> TimeSeriesDataFrame:

        # Generate target predictions
        known_covariates_cp = known_covariates.copy() if known_covariates is not None else None
        all_preds = []
        for target in self.target_columns:
            print(f"Predicting target: {target}")
            predictor = self.get_predictor(target)

            # Get raw quantile predictions. Passing through the quantile_levels arg will not work since quantiles have been hard-encoded in models
            raw_pred = predictor.predict(
                data,
                known_covariates=known_covariates_cp,
                **predict_kwargs
            )

            # Format result
            target_df = pd.DataFrame(index=raw_pred.index)

            # Return requested quantiles used in logic
            for q in quantile_levels:
                if str(q) in raw_pred.columns:
                    target_df[f"{target}_{q}"] = raw_pred[str(q)]

            all_preds.append(target_df)

        final_df = pd.concat(all_preds, axis=1)
        static_features = getattr(data, 'static_features', None)
        return TimeSeriesDataFrame(final_df, static_features=static_features)


    def evaluate(self, data: TimeSeriesDataFrame, **kwargs) -> dict:
        """
        Returns a dictionary of evaluation scores, one for each target.

        Evaluation is always performed independently for each model against the ground truth.
        The chained predictions from `predict` are not used here, as that would be incorrect.
        """
        if not isinstance(data, TimeSeriesDataFrame):
            data = TimeSeriesDataFrame(data)

        eval_dict = {}
        for target in self.target_columns:
            print(f"Evaluating predictor for target: {target} ...")
            predictor = self.get_predictor(target)
            eval_dict[target] = predictor.evaluate(data, **kwargs)

        return eval_dict


    def save(self):
        """
        Saves the MultiTargetTimeSeriesPredictor to disk.

        This method follows the only sane pattern: it saves each individual predictor
        and then saves the wrapper object with PATHS to the predictors, not the objects
        themselves. This is how the Tabular example you provided works, and it's the right way.
        """
        # Replace predictor objects with their paths before saving the wrapper
        for target, predictor in self.predictors.items():
            if not isinstance(predictor, str):
                self.predictors[target] = predictor.path

        save_path = os.path.join(self.path, self._PREDICTOR_FILENAME)
        save_pkl.save(path=save_path, object=self)
        print(f"MultiTargetTimeSeriesPredictor saved to disk. Load with: MultiTargetTimeSeriesPredictor.load('{self.path}')")

    @classmethod
    def load(cls, path: str):
        """Loads a MultiTargetTimeSeriesPredictor from the specified path."""
        load_path = os.path.join(path, cls._PREDICTOR_FILENAME)
        if not os.path.exists(load_path):
             raise FileNotFoundError(f"Cannot find predictor pickle file at {load_path}")
        return load_pkl.load(path=load_path)

    def get_predictor(self, target: str) -> TimeSeriesPredictor:
        """
        Returns the underlying TimeSeriesPredictor for a specific target.

        If the predictor is referenced by its path (e.g., after loading from disk),
        it will be loaded into memory. This is the on-demand loading pattern.
        """
        if target not in self.predictors:
            raise ValueError(f"Unknown target: {target}. Available targets: {list(self.predictors.keys())}")

        predictor_or_path = self.predictors[target]
        if isinstance(predictor_or_path, str):
            # Load from path and update the dictionary so we don't have to load it again in this session
            loaded_predictor = TimeSeriesPredictor.load(predictor_or_path)
            self.predictors[target] = loaded_predictor
            return loaded_predictor

        return predictor_or_path

baseline_predictor = MultiTargetTimeSeriesPredictor(all_targets,
                                                    f"AutogluonModels/{version_name}_MTSPredictor",
                                                    eval_metric=eval_metric,
                                                    predictor_init_kwargs={
                                                        'prediction_length': 1,
                                                        'freq': 'D',
                                                        'known_covariates_names' : [str(col) for col in train_ts_df.columns
                                                                                    if col not in all_targets and col != "timestamp" and col != "item_id"]
                                                    })

baseline_predictor.fit(train_ts_df,
                       hyperparameters=hyperparameters,
                       ensemble_hyperparameters=ensemble_hyperparameters,
                       enable_ensemble=enable_ensemble,
                       num_val_windows=backtesting_times_in_val)
```

**Screenshots / Logs**
```
...(passed)
Fitting 1 ensemble(s), in 1 layers.
Training ensemble model TabularEnsemble. 
	Warning: Exception caused TabularEnsemble to fail during training... Skipping this model.
	cannot reshape array of size 429000 into shape (14300,)
```
We have 29 covariate columns and 1 target column at this fitting, i.e. 30 columns, which is equals to 429000/14300.

**Installed Versions**
```
autogluon              : 1.5.0
autogluon.common       : 1.5.0
autogluon.core         : 1.5.0
autogluon.features     : 1.5.0
autogluon.multimodal   : 1.5.0
autogluon.tabular      : 1.5.0
autogluon.timeseries   : 1.5.0
boto3                  : 1.40.64
catboost               : 1.2.8
chronos-forecasting    : 2.2.2
coreforecast           : 0.0.16
defusedxml             : 0.7.1
einops                 : 0.8.1
einx                   : 0.3.0
evaluate               : 0.4.6
fastai                 : 2.8.5
fsspec                 : 2025.3.0
fugue                  : 0.9.1
gluonts                : 0.16.2
huggingface_hub        : 0.36.0
hyperopt               : 0.2.7
imodels                : None
interpret-core         : None
jinja2                 : 3.1.6
joblib                 : 1.5.2
jsonschema             : 4.23.0
lightgbm               : 4.6.0
lightning              : 2.5.5
loguru                 : 0.7.3
matplotlib             : 3.10.7
mlforecast             : 0.14.0
networkx               : 3.5
nlpaug                 : 1.1.11
nltk                   : 3.9.2
numpy                  : 2.1.3
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.3.0
onnx                   : None
onnxruntime            : None
onnxruntime-gpu        : None
openmim                : 0.3.9
orjson                 : 3.11.4
pandas                 : 2.3.3
pdf2image              : 1.17.0
peft                   : 0.17.1
Pillow                 : 11.3.0
psutil                 : 7.0.0
pyarrow                : 20.0.0
pytabkit               : 1.7.0
pytesseract            : 0.3.13
pytorch-metric-learning: 2.8.1
pyyaml                 : 6.0.2
ray                    : 2.44.1
requests               : 2.32.5
scikit-image           : 0.25.2
scikit-learn           : 1.7.2
scikit-learn-intelex   : None
scipy                  : 1.15.3
seqeval                : 1.2.2
skl2onnx               : None
spacy                  : 3.8.7
statsforecast          : 2.0.1
stevedore              : 5.4.1
tabdpt                 : None
tabicl                 : None
tabpfn                 : None
tensorboard            : 2.20.0
text-unidecode         : 1.3
timm                   : 1.0.3
torch                  : 2.7.1+cu128
torchmetrics           : 1.7.4
torchvision            : 0.22.1+cu128
tqdm                   : 4.67.1
transformers           : 4.57.3
utilsforecast          : 0.2.11
xgboost                : 3.0.5
```

**Comment by JulienMoisan** [comment]
In case helpful, here is a smaller repro - the below no longer throws with the PR I made above.

```python
from autogluon.timeseries import TimeSeriesPredictor, TimeSeriesDataFrame
import pandas as pd
import numpy as np

# Create a simple synthetic dataset with a known covariate
np.random.seed(42)
num_items = 5
num_timestamps = 100

rows = []
for item_id in range(num_items):
    timestamps = pd.date_range("2020-01-01", periods=num_timestamps, freq="D")
    values = np.random.randn(num_timestamps).cumsum()
    for t, v in zip(timestamps, values):
        rows.append({"item_id": item_id, "timestamp": t, "volume": v, "known_covariate": 1})

df = pd.DataFrame(rows)
train_data = TimeSeriesDataFrame.from_data_frame(df, id_column="item_id", timestamp_column="timestamp")

predictor = TimeSeriesPredictor(
    target="volume",
    prediction_length=5,
    known_covariates_names=["known_covariate"],
)

# This will log:
#   "Warning: Exception caused TabularEnsemble to fail during training... Skipping this model."
#   "cannot reshape array of size 100 into shape (50,)"
predictor.fit(
    train_data,
    hyperparameters={"Naive": {}, "SeasonalNaive": {}},
    ensemble_hyperparameters={"TabularEnsemble": {}},
    num_val_windows=2,
    enable_ensemble=True,
)
```
